code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -17,6 +17,20 @@
 	on_reagent_change()
 		update_icon()
 
+		if(volume == 0 && name != "Empty Bloodback")
+			name = "Empty Bloodpack"
+			desc = "Empty bloodpacks are good in vampire movies, but bad in hospitals."
+		else if (reagents.reagent_list.len > 0)
+			var/target_type = null
+			var/the_volume = 0
+			for(var/datum/reagent/A in reagents.reagent_list)
+				if(A.volume > the_volume && ("blood_type" in A.data))
+					the_volume = A.volume
+					target_type = A.data["blood_type"]
+			name = "[target_type] Bloodpack"

```

Comment:
I don't think there's a way to get things other than blood into a bloodpack, because it is not an open container. According to the IRC, tests have been done and no one has been able to get other stuff in.

I guess it wouldn't hurt to just throw it in though.